canon_host/tests/root_test.rs

```diff
@@ -0,0 +1,191 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+#![feature(min_const_generics)]
+
+use canonical_host::{
+    wasm, Apply, CastMut, Execute, MemStore, Query, Remote, Root, Transaction,
+};
+
+use canonical::{Canon, Store};
+use canonical_derive::Canon;
+
+use counter::{self, Counter};
+
+type ContractAddr = usize;
+
+// transactions
+const ADD_CONTRACT: u8 = 0;
+const APPLY_CONTRACT_TRANSACTION: u8 = 1;
+
+// queries
+const EXECUTE_CONTRACT_QUERY: u8 = 0;
+
+#[derive(Clone, Canon, Default)]
+struct TestState<S: Store> {
+    opcount: u64,
+    contracts: Vec<Remote<S>>,
+}
```

Comment:
I'm unsure how this is going to work once we integrate it. 

Does this mean that we will need to append to this structure all of the contracts? Is this a structure just created to generate delagate calls?